### PR TITLE
Move email templates

### PR DESCRIPTION
**Pull request by akellbl4**

### CODE REVIEW

File: backend/vendor/github.com/go-pkgz/auth/auth.go [hunk 1]
```diff
@@ -266,6 +266,7 @@ func (s *Service) AddDirectProvider(name string, credChecker provider.CredChecke
 
 // AddVerifProvider adds provider user's verification sent by sender
 func (s *Service) AddVerifProvider(name, msgTmpl string, sender provider.Sender) {
+
```
Comment (paskal) [inline backend/vendor/github.com/go-pkgz/auth/auth.go]: Please revert changes to vendor:)

File: backend/app/templates/email_reply.html.tmpl [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+					<span style="font-size: 14px; font-weight: bold; color: #777">{{.UserName}}</span>
+					<span style="color: #999; font-size: 14px; margin: 0 8px;">{{.CommentDate.Format "02.01.2006 at 15:04"}}</span>
+					<a href="{{.CommentLink}}" style="color: #0aa; font-size: 14px;"><b>Reply</b></a>
+				</div>
+				<div style="font-size: 16px; background-color: #fff; color:#000!important; padding: 14px 14px 2px 14px; border-radius: 3px; line-height: 1.4;">{{.CommentText}}</div>
+			</div>
+		</div>
+		<div style="text-align: center; font-size: 14px; margin-top: 32px;">
+			<i style="color: #000!important;">Sent to <a style="color:inherit; text-decoration: none" href="mailto:{{.Email}}">{{.Email}}</a>{{if not .ForAdmin}} for {{.ParentUserName}}{{ end }}</i>
+			<div style="margin: auto; width: 150px; border-top: 1px solid rgba(0, 0, 0, 0.15); padding-top: 15px; margin-top: 15px;"></div>
```
Comment (paskal) [inline backend/app/templates/email_reply.html.tmpl]:
```suggestion
			<div style="width: 150px; border-top: 1px solid rgba(0, 0, 0, 0.15); padding-top: 15px; margin: 15px auto auto;"></div>
```

File: backend/app/templates/email_reply.html.tmpl [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			{{- if .UnsubscribeLink}}
+			<a style="color: #0aa;" href="{{.UnsubscribeLink}}">Unsubscribe</a>
+			{{- end }}
+			<!-- This is hack for remove collapser in Gmail which can collapse end of the message -->
+			<div style="opacity: 0;font-size: 1;">[{{.CommentDate.Format "02.01.2006 at 15:04"}}]</div>
```
Comment (paskal) [inline backend/app/templates/email_reply.html.tmpl]: > `font-size: 1`

Something is wrong there.

File: backend/app/templates/email_reply.html.tmpl [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+				<span style="font-size: 14px; font-weight: bold; color: #777">{{.ParentUserName}}</span>
+				<span style="color: #999; font-size: 14px; margin: 0 8px;">{{.ParentCommentDate.Format "02.01.2006 at 15:04"}}</span>
+				<a href="{{.ParentCommentLink}}" style="color: #0aa; font-size: 14px;"><b>Show</b></a>
+			</div>
+			<div style="font-size: 14px; color:#333!important; padding: 0 14px 0 2px; border-radius: 3px; line-height: 1.4;">
+				{{.ParentCommentText}}
+			</div>
+						{{- end }}
+			<div style="padding-left: 20px; border-left: 1px dotted rgba(0,0,0,0.15); margin-top: 15px; padding-top: 5px;">
+				<div style="margin-bottom: 12px;" line-height: 24px;word-break: break-all;>
```
Comment (paskal) [inline backend/app/templates/email_reply.html.tmpl]:
```suggestion
				<div style="margin-bottom: 12px; line-height: 24px;word-break: break-all;">
```

File: backend/app/templates/templates.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+type FileSystem interface {
+	ReadFile(name string) ([]byte, error)
+}
+
+type osFS struct{}
+
+var statikInstance http.FileSystem
+var isStatik bool
+
+func Init() FileSystem {
```
Comment (paskal) [inline backend/app/templates/templates.go:29]: We don't do that in Go. It's NewFileSystem, not Init, and it should return pointer.

File: backend/app/templates/templates.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	Stat() (os.FileInfo, error)
+}
+type FileSystem interface {
+	ReadFile(name string) ([]byte, error)
+}
+
+type osFS struct{}
+
+var statikInstance http.FileSystem
+var isStatik bool
```
Comment (paskal) [inline backend/app/templates/templates.go:27]: These things should be embedded into osFS, no global variables please.

File: backend/app/cmd/server.go [hunk 7]
```diff
@@ -96,9 +96,9 @@ type ServerCommand struct {
 			Port         int           `long:"port" env:"PORT" description:"[deprecated, use --smtp.port] SMTP password"`
 			SMTPPassword string        `long:"passwd" env:"PASSWD" description:"[deprecated, use --smtp.password] SMTP port"`
 			SMTPUserName string        `long:"user" env:"USER" description:"[deprecated, use --smtp.username] enable TLS"`
-			TLS          bool          `long:"tls" env:"TLS" description:"[deprecated, use --smtp.tls] SMTP TCP connection timeout"`
-			TimeOut      time.Duration `long:"timeout" env:"TIMEOUT" default:"10s" description:"[deprecated, use --smtp.timeout] SMTP TCP connection timeout"`
-			MsgTemplate  string        `long:"template" env:"TEMPLATE" description:"message template file"`
+			TLS               bool          `long:"tls" env:"TLS" description:"[deprecated, use --smtp.tls] SMTP TCP connection timeout"`
```
Comment (paskal) [inline backend/app/cmd/server.go]: This part is not formatted using `gofmt -s -w .`, it should be.

File: backend/app/cmd/server.go [hunk 8]
```diff
@@ -363,8 +365,8 @@ func (s *ServerCommand) newServerApp() (*serverApp, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to make avatar store")
 	}
-	authenticator := s.makeAuthenticator(dataService, avatarStore, adminStore)
 
+	authenticator := s.makeAuthenticator(dataService, avatarStore, adminStore)
```
Comment (paskal) [inline backend/app/cmd/server.go]: unnecessary changes

File: backend/app/cmd/server.go [hunk 9]
```diff
@@ -653,30 +655,7 @@ func (s *ServerCommand) makeCache() (LoadingCache, error) {
...
-		<p style="position: relative; font-family: monospace; background-color: #fff; margin: 0; padding: 0.5em; word-break: break-all; text-align: left; border-radius: 0.2em; -webkit-user-select: all; user-select: all;">{{.Token}}</p>
-	</div>
-	<p style="position: relative; margin-top: 2em; font-size: 0.8em; opacity: 0.8;"><i>Sent to {{.Address}}</i></p>
-</div>
-</body>
-</html>
-`
-
 func (s *ServerCommand) addAuthProviders(authenticator *auth.Service) {
-
```
Comment (paskal) [inline backend/app/cmd/server.go:679]: unnecessary changes

File: backend/app/notify/email.go [hunk 10]
```diff
@@ -9,22 +9,24 @@ import (
...
-	From                 string // from email address
-	MsgTemplate          string // request message template
-	VerificationSubject  string // verification message subject
-	VerificationTemplate string // verification message template
-	SubscribeURL         string // full subscribe handler URL
-	UnsubscribeURL       string // full unsubscribe handler URL
+	From                     string // from email address
+	VerificationSubject      string // verification message sub
+	VerificationTemplatePath string // path to verification message template
+	ReplyTemplatePath        string // path to reply message template
```
Comment (paskal) [inline backend/app/notify/email.go:27]: I think "message template" is clearer name than "reply template". Discussable, I want to hear your logic behind it.

File: backend/app/notify/email.go [hunk 11]
```diff
@@ -295,8 +212,8 @@ func (e *Email) buildVerificationMessage(user, email, token, site string) (strin
 	return e.buildMessage(subject, msg.String(), email, "text/html", "")
 }
 
-// buildMessageFromRequest generates email message based on Request using e.MsgTemplate
-func (e *Email) buildMessageFromRequest(req Request, forAdmin bool) (string, error) {
+// buildReplyRequest generates email message based on Request using e.replyTmpl
```
Comment (paskal) [inline backend/app/notify/email.go]: It's building a message from the request, not reply request. I would suggest undoing renames if you have no strong feelings about them, as new ones don't feel better than old ones.

**Review by paskal — CHANGES_REQUESTED** [review]

File: backend/app/notify/email.go (hunk 10 above)
Comment (akellbl4) [reply backend/app/notify/email.go:27]: I changed it because it is not template it is a just path to file and the template should be read by it but not parsed.

Comment (paskal) [reply backend/app/notify/email.go:27]: I propose to use "msgTemplatePath" then.

File: backend/app/notify/email.go (hunk 11 above)
Comment (akellbl4) [reply backend/app/notify/email.go]: My logic in this:
- we have two types of emails in our service
- first for verification and second for what?
- this template notifies users/admins about replies

1. it is a reply to a user comment
2. it is a reply to an admin article on site

P.s. I made wrong naming I wanted to name it like `buildReplyMessage`
I haven't understood why this email builds by request.

Comment (paskal) [reply backend/app/notify/email.go]: `buildReplyMessage` would be appropriate name

File: backend/app/notify/email.go (hunk 10 above)
Comment (akellbl4) [reply backend/app/notify/email.go:27]: 1. verify template is ok, it is a message for verification you subscription
2. message? for what? because my opinion to keep it `reply` because it is much clear for what this template is using. Maybe I am wrong, maybe I should understand why it should be just a message template.

Could you give me your logic in naming?

Comment (paskal) [reply backend/app/notify/email.go:27]: No problem with current name if the function below will get correct name as well.

File: README.md [hunk 12]
```diff
@@ -156,6 +156,8 @@ _this is the recommended way to run remark42_
 | notify.email.fromAddress | NOTIFY_EMAIL_FROM      |                          | from email address                              |
 | notify.email.verification_subj | NOTIFY_EMAIL_VERIFICATION_SUBJ | `Email verification` | verification message subject          |
 | notify.email.notify_admin | NOTIFY_EMAIL_ADMIN    | `false`                  | notify admin on new comments via ADMIN_SHARED_EMAIL |
+| notify.email.verification_tmpl_path | NOTIFY_EMAIL_VERIFICATION_TMPL_PATH | `./templates/email_verification.html.tmpl` | path to verification email template |
```
Comment (paskal) [inline README.md:159]: I think we can leave them unchangable, just document where to take them. What do you think?

File: backend/app/templates/templates.go (hunk 6 above)
Comment (paskal) [reply backend/app/templates/templates.go:27]: Point still stays the same, there should be no global variables.

**Review by paskal — CHANGES_REQUESTED** [review]

File: README.md (hunk 12 above)
Comment (akellbl4) [reply README.md:159]: I also don't like this long name of var.
